hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/BucketManagerImpl.java

```diff
@@ -114,9 +115,11 @@ public void createBucket(OmBucketInfo bucketInfo) throws IOException {
     Preconditions.checkNotNull(bucketInfo);
     String volumeName = bucketInfo.getVolumeName();
     String bucketName = bucketInfo.getBucketName();
-    metadataManager.getLock().acquireVolumeLock(volumeName);
-    metadataManager.getLock().acquireBucketLock(volumeName, bucketName);
+    boolean acquiredBucketLock = false;
+    metadataManager.getLock().acquireLock(VOLUME_LOCK, volumeName);
```

Comment:
There is a subtle bug here. We either need to assume that acquireLock cannot fail, or assume that it can fail. 
If it cannot fail: Then the finally is simple, you can release bucket and volume lock.
if it can fail: Then it is possible the acquire volume lock is going to fail -- and there is no handling -- I am presuming that it will throw up into the calling unction and there is some sort of a handler there ? if that is the case,  what will it throw ? I am not sure it can fail --  @arp7  comments?

Reply:
We have to assume that it can fail for two reasons:

1. Because of the following check:
```
  public void acquireVolumeLock(String volume) {
    // Calling thread should not hold any bucket lock.
    // You can take an Volume while holding S3 bucket lock, since
    // semantically an S3 bucket maps to the ozone volume. So we check here
    // only if ozone bucket lock is taken.
    if (hasAnyBucketLock()) {
      throw new RuntimeException(
          "Thread '" + Thread.currentThread().getName() +
              "' cannot acquire volume lock while holding bucket lock(s).");
    }
```
1. The wrapped call to `lockPool.borrowObject()` in `LockManager#lock` declares that it can throw `Exception`. So that covers pretty much any exception under the earth.

Reply:
If lock acquire fails, we get RunTimeException.

Reply:
Looked at this further. Failure to get volume lock is fine here. It is up to the caller to release their resources when the stack is unwound.

Reply:
I haven't looked at the entire patch however I am +1 on this particular point.